tools/lib/template_parser.py

```diff
@@ -236,6 +236,21 @@ def is_django_block_tag(tag):
         'raw',
     ]
 
+def extract_line(text, offset):
+    # type: (str, int) -> Dict[str, Any]
+    details = {} # type: Dict[str, Any]
+    end_of_line = offset
+    while text[end_of_line] != '\n':
+        end_of_line += 1
+
+    details['lookup'] = text[offset:end_of_line]
+    details['line_num'] = 1
+    for line in text.split('\n'):
```

Comment:
Since we already know the offset here, I think we should create a function called line_number_of_offset(offset) and say `details['line_num'] = line_number_of_offset()`